config.def.h

```diff
@@ -163,7 +163,7 @@ static const keymap_t keys[] = {
 };
 
 /* mouse button mappings for image mode: */
-static const button_t buttons[] = {
+static const button_t buttons_img[] = {
```

Comment:
Why not just merge them into a single array like we have for normal bindings?

Reply:
It wouldn't work due to fallthough. For example M3 would switch mode and then end up selecting an image. We could maybe get around it by forcing all the thumbnail bindings to be on top.

Reply:
Fair point. I would say that sounds like a trap, but imagine the same situation is possible for keybindings. We could save the mode at the start of process_bindings so that we aren't affected if it changes. 

Putting the thumbnails on top also is good enough to avoid the issue.

Reply:
> We could save the mode at the start of process_bindings so that we aren't affected if it changes.

That could lead to some unforeseen consequences with keybindings. I think a separate array for thumbnails is OK.

What I was more concerned with is the rename of `buttons` to `buttons_img`, it should be a trivial change for people upgrading their config.h so I decided that appending `_img` to it is ok. But yeah, it does create a small merge conflict when upgrading, but keeping the old name makes the array naming inconsistent.

Reply:
>That could lead to some unforeseen consequences with keybindings.

Which we don't care about because we don't consider custom configs and the default behavior could be the same.

>I think a separate array for thumbnails is OK.

I don't really like the asymmetry between buttons and keypresses. To me a main benefit of this patch is uniformity and don't want to lose that in the most visible place. Would rather just stick the thumbnails on top then have 2 arrays. A downside of this is that existing source users with a custom config may just lose thumbnail bindings without immediately realizing it because unlike other config changes, this wouldn't break their config and thereby notify them something changed. But I don't think this is a real concern.

Reply:
> Which we don't care about because we don't consider custom configs and the default behavior could be the same.

We don't care about supporting custom functions. We do care about keeping the keybinding processing logic same, if we didn't then #176 doesn't make sense either since all default keybinding work fine.